var/spack/repos/builtin/packages/openfoam-com/package.py

```diff
@@ -236,15 +236,19 @@ def mplib_content(spec, pre=None):
     else:
         pre = mpi_spec.prefix
 
+    libname = 'mpi'
+    if 'mpich' in mpi_spec.name:
+        libname = 'mpich'
+
     info = {
         'name':   '{0}-{1}'.format(mpi_spec.name, mpi_spec.version),
         'prefix':  pre,
         'include': inc,
         'bindir':  bin,
         'libdir':  lib,
-        'FLAGS':  '-DOMPI_SKIP_MPICXX -DMPICH_IGNORE_CXX_SEEK',
+        'FLAGS':  '-DOMPI_SKIP_MPICXX -DMPICH_SKIP_MPICXX',
         'PINC':   '-I{0}'.format(inc),
-        'PLIBS':  '-L{0} -lmpi'.format(lib),
+        'PLIBS':  '-L{0} -l{1}'.format(lib, libname),
```

Comment:
With the `pre` I'm still trying to play both fields: use all of the specifications from spack, but also retain the OpenFOAM-style of having relocatable packages. It doesn't quite fit together, but that's where we are at. The MPI handling in OpenFOAM is encapsulated in its Pstream library, so it is possible to recompile several of these (for various MPI implementations) and swap them out by adjusting the LD_LIBRARY_PATH.

This is pretty much the diametrical opposite of spack, so we are mostly glad that we even get things working together at all.